### PR TITLE
[MM-48649]: Call the SendSubscriptionHistoryEvent function in a GO routine

### PR DESCRIPTION
**Pull request by emmyni**
#### Summary

The `SendSubscriptionHistoryEvent` function will now be called in a Go routine, so that it does not stall the UI when a new user is created.

Also related to this PR: https://github.com/mattermost/enterprise/pull/1317
#### Ticket Link

https://mattermost.atlassian.net/browse/MM-48649
#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by neallred — APPROVED** [review]
LGTM, thanks @emmyni !

**Comment by emmyni** [comment]
This is the same as this PR: https://github.com/mattermost/mattermost-server/pull/21813. I didn't give the closed PR's branch and the enterprise PR's branch the same name originally, so it was having trouble syncing up the changes. Additionally, there was another problem with the enterprise master branch originally which caused it to fail some checks. I couldn't get it to pass the checks for some reason, so I had to make a new branch for the changes.

**Review by ConorMacpherson — APPROVED** [review]
LGTM :)

File: app/user.go [hunk 1]
```diff
@@ -316,10 +316,13 @@ func (a *App) createUserOrGuest(c request.CTX, user *model.User, guest bool) (*m
 		}, plugin.UserHasBeenCreatedID)
 	})
 
-	_, cwsErr := a.SendSubscriptionHistoryEvent(ruser.Id)
-	if cwsErr != nil {
-		c.Logger().Error("Failed to create/update the SubscriptionHistoryEvent", mlog.Err(cwsErr))
-	}
+	// Create/Update the subscriptionHistoryEvent
+	go func() {
```
Comment (AGMETEOR) [inline app/user.go:320]: We needed to add a comment here why this is part of the user creation process

Comment (AGMETEOR) [reply app/user.go:320]: @emmyni should there also be a check for cloud?

**Pull request merged**